### ISSUE TITLE
'TypeError: this.getPosition is not a function' on parsing inline comments

### ISSUE THREAD

**Issue by FreekVR** [shellscape/postcss-less#165]
* Node Version: 14.8.1
* NPM Version: 6.14.14
* postcss Version: 8.3.11
* postcss-less Version: 5.0.0

### LESS

```less
.layout.layout--auth {
    display: flex;
    justify-content: center;
    min-height: 100vh;
    background: #000285 url('~assets/img/backgrounds/auth-bg.jpg'); // @TODO: Make responsive & smaller where possible
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;

    @media @q-sm-min {
        align-items: center;
    }
}
```

### Errors 

```
TypeError: this.getPosition is not a function
    at LessParser.inlineComment (/home/freek/projects/asc-app/node_modules/postcss-less/lib/LessParser.js:72:28)
    at LessParser.isInlineComment (/home/freek/projects/asc-app/node_modules/postcss-less/lib/nodes/inline-comment.js:37:12)
    at LessParser.other (/home/freek/projects/asc-app/node_modules/postcss-less/lib/LessParser.js:157:36)
    at LessParser.parse (/home/freek/projects/asc-app/node_modules/postcss/lib/parser.js:75:16)
    at Object.parse (/home/freek/projects/asc-app/node_modules/postcss-less/lib/index.js:11:12)
    at LocalFixer.parse (/home/freek/projects/asc-app/node_modules/postcss-html/lib/html/parse-styles.js:96:18)
    at parseStyle (/home/freek/projects/asc-app/node_modules/postcss-html/lib/html/parse-styles.js:125:43)
    at /home/freek/projects/asc-app/node_modules/postcss-html/lib/html/parse-styles.js:138:18
    at Array.forEach (<anonymous>)
    at parseStyles (/home/freek/projects/asc-app/node_modules/postcss-html/lib/html/parse-styles.js:137:5)
```

### Expected Behavior
No error, parser succeeds

### Actual Behavior
Getting an error, could not even find a function with that name in any of the postcss-less files :(

**Comment by jwbrandon** [comment]
This happened to us because of an out-of-date version of postcss.

**Comment by nwalters512** [comment]
It looks like `postcss-less` relies on v8 of `postcss`, but this isn't declared in `package.json`. In my case, this meant that Yarn had hoisted v7 of `postcss`, which is what `postcss-less` ultimately required. @shellscape would you accept a PR adding `"postcss": "^8.3.5"` to `dependencies` in `package.json` to match what's in `devDependencies` at the moment?

**Comment by shellscape** [comment]
@nwalters512 absolutely, but make it a peerDep.

**Comment by nwalters512** [comment]
@shellscape Unfortunately I don't think a peer dep would help here, as Yarn doesn't actually install them (I know recent versions of npm will, but we can't universally rely on that). Is there a specific reason you want to avoid making it an actual dependency?

**Comment by shellscape** [comment]
Bloat, mostly. This is an incredibly common practice these days. yarn _will_ warn when a peer dep doesn't match or isn't installed. As will pnpm.  If you're using postcss v7 directly and relying on a tool that uses v8, youve got some dep tree problems there.

**Comment by nwalters512** [comment]
The thing is that I'm actually not installing `postcss` directly at all. We rely on a lot of deps that in turn rely on `postcss`, such as `stylelint`, which does [list it in `dependencies`](https://github.com/stylelint/stylelint/blob/71b7ccb18aa5e23ac58da44527fe865ecde05b87/package.json#L135). Unfortunately, a lot of *other* tools we use depend on v7, which Yarn then chooses to hoist to the root of `node_modules`. Yarn doesn't know that that's actually a problem because `postcss-less` doesn't explicitly depend on it, and AFAICT Yarn doesn't take peer deps into account when deciding what to hoist.

Re: bloat, adding it as a dependency shouldn't cause bloat, as it *is* actually a runtime dependency of this library - users would have already had to have installed it for this to work correctly at all. Yarn and npm will dedupe duplicate dependencies, so this shouldn't result in an additional copy of `postcss` being installed either.

**Comment by shellscape** [comment]
I'm sorry you're having hoisting issues with yarn, but peeDeps is the correct fix in my opinion.

**Comment by nwalters512** [comment]
AFAIK this won't actually fix things for consumers, as there's not actually any way to force Yarn to reliably install the *correct* version of that peer dependency so that it's visible to `postcss-less`.

**Comment by shellscape** [comment]
6.5mil downloads a week would disagree with you https://github.com/shellscape/postcss-values-parser/blob/5aea4d7e1b5cc8ebb9401379ef06c17213585493/package.json#L56

**Comment by nwalters512** [comment]
The fact that this issue was opened means that this is a problem that should be fixed, regardless of how many downloads a package has.

**Comment by shellscape** [comment]
welp. you have my response. you're welcome to open a PR to that effect, wait for me to get around to an update myself, or fork if you disagree with my take on it. All are acceptable.

**Comment by nwalters512** [comment]
👍 we'll inline this package into our codebase!

**Comment by shellscape** [comment]
fixed in 312a9a12d59f0ac8a7fe0e42bc31e8bb66e4630b
